src/lib.rs

```diff
@@ -626,6 +626,105 @@ where
         };
     }
 
+    /// Return the number of contiguous clusters. If the next cluster in the sequence isn't contiguous
+    /// (i.e. is fragmented), it returns `1`
+    fn check_contiguous_cluster_count(
+        &self,
+        volume: &Volume,
+        mut cluster: Cluster,
+    ) -> Result<u32, Error<D::Error>> {
+        let mut contiguous_cluster_count = 1u32;
+        let mut next_cluster = match &volume.volume_type {
+            VolumeType::Fat(fat) => match fat.next_cluster(self, cluster) {
+                Ok(cluster) => cluster,
+                Err(e) => match e {
+                    // If this is the last cluster for the file, simply return the same cluster.
+                    Error::EndOfFile => cluster,
+                    _ => panic!(
+                        "Error: traversing the FAT table, accessed free space or a bad cluster"
+                    ),
+                },
+            },
+        };
+        while (next_cluster.0 - cluster.0) == 1 {
```

Comment:
I'm not well-versed with the FAT standard, so I may be wrong here but I believe contiguous clusters are (always) in ascending order i.e. if we were to come across a `next cluster` that precedes the `current cluster`, its considered to be a fragmented cluster (and we can simply return 1). 

But in any case, block devices that support `multiple block reads` require a `starting block` and a `block-count`. The implementation can get tricky if we want to account for `next clusters` preceding the current one. So, to avoid additional complexity, I simply assume they are fragmented.

Reply:
Right but if cluster.0 is 4 and next_cluster.0 is 3, this subtraction will overflow before we check == 1. Right?

Reply:
Ah, I understand now (was overthinking the question, got carried away with the FAT stuff). Yes, that would result in an overflow. I've updated it to use `wrapping_sub`